README.md

```diff
@@ -7,101 +7,117 @@
 ```
 
-To watch the process, `juju status` can be used. Once all the units show as `active|idle` the credentials to access a broker can be queried with:
+To watch the process, the `juju status` command can be used. Once all the units shown as `active|idle`, the credentials to access a broker can be queried with:
+
 ```shell
 juju run-action kafka-k8s/leader get-admin-credentials --wait 
 ```
 
-### Replication
-#### Scaling application
-The charm can be scaled using `juju scale-application` command.
+### Scaling
+
+The charm can be scaled using `juju scale-application` command:
+
 ```shell
-juju scale-application kafka-k8s <num_of_brokers_to_scale_to>
+juju scale-application kafka-k8s <num_of_units_to_scale_to>
 ```
 
-This will add or remove brokers to match the required number. To scale a deployment with 3 kafka units to 5, run:
+This will add or remove brokers to match the required number. For example, to scale a deployment with 3 kafka units to 5, run:
+
 ```shell
 juju scale-application kafka-k8s 5
 ```
 
 ### Password rotation
-#### Internal operator user
-The operator user is used internally by the Charmed Kafka K8s Operator, the `set-password` action can be used to rotate its password.
+
+The operator user is used internally by the Charmed Kafka K8s Operator. The `set-password` action can be used to rotate its password:
+
 ```shell
-# to set a specific password for the operator user
 juju run-action kafka-k8s/leader set-password password=<password> --wait
-
-# to randomly generate a password for the operator user
-juju run-action kafka-k8s/leader set-password --wait
 ```
 
+Use the same action without a password parameter to randomly generate a password for the operator user.
+
 ### Storage support
 
-Currently, the Charmed Kafka K8s Operator makes use of a 10GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
+Currently, the Charmed Kafka K8s Operator makes use of a 10 GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
+
+This storage is mounted on `/var/lib/data/kafka` and used for log-data.
 
-This is used for logs storage, mounted on `/logs/kafka`
+Service logs can be found in `/var/log/kafka`.
 
 ## Relations
 
-Supported [relations](https://juju.is/docs/olm/relations):
+The Charmed Kafka K8s Operator supports Juju [relations](https://juju.is/docs/olm/relations) for interfaces listed below.
 
-#### `kafka_client` interface:
+#### The kafka_client interface
 
-The `kafka_client` interface is used with the `data-integrator` charm. This charm allows to automatically create and manage product credentials needed to authenticate with different kinds of data platform charmed products:
+The `kafka_client` interface is used with the `data-integrator` charm. This charm allows to automatically create and manage product credentials needed to authenticate with different kinds of data platform charmed products.
```

Comment:
```suggestion
The `kafka_client` interface is used with the `data-integrator` charm, which upon relation automatically provides credentials and endpoints for connecting to the desired product.
```

I would add a link to the Data Integrator charm

Reply:
```suggestion
The `kafka_client` interface is used with the [Data Integrator](https://charmhub.io/data-integrator) charm, which upon relation automatically provides credentials and endpoints for connecting to the desired product.
```